src/test/java/net/fabricmc/test/TestMod.java

```diff
@@ -17,16 +17,31 @@
 package net.fabricmc.test;
 
 import net.fabricmc.api.ModInitializer;
-import net.fabricmc.loader.FabricLoader;
+import net.fabricmc.loader.api.entrypoint.PreMainEntrypoint;
+import net.fabricmc.loader.launch.common.FabricLauncherBase;
+
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-public class TestMod implements ModInitializer {
+public class TestMod implements PreMainEntrypoint, ModInitializer {
 
 	private static final Logger LOGGER = LogManager.getFormatterLogger("TestMod");
 
+	@Override
+	public void onPreMain() {
```

Comment:
Consider splitting this into another class so as to achieve asm safety (i.e. game entrypoints are not classloaded with prelaunch entry points)

Reply:
It is only the test mod, I'll add a warning to not do this in production though